modules/gapi/include/opencv2/gapi/streaming/onevpl/onevpl_data_provider_interface.hpp

```diff
@@ -0,0 +1,42 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ONEVPL_DATA_PROVIDER_INTERFACE_HPP
+#define GAPI_STREAMING_ONEVPL_ONEVPL_DATA_PROVIDER_INTERFACE_HPP
+#include <exception>
+#include <string>
+
+#include <opencv2/gapi/own/exports.hpp> // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+struct GAPI_EXPORTS DataProviderException : public std::exception {
+    virtual ~DataProviderException() {};
+};
+
+struct GAPI_EXPORTS DataProviderSystemErrorException : public DataProviderException {
+    DataProviderSystemErrorException(int error_code, const std::string& desription = std::string());
+    virtual ~DataProviderSystemErrorException();
+    virtual const char* what() const noexcept override;
+
+private:
+    std::string reason;
+};
+
+struct GAPI_EXPORTS IDataProvider {
+    using Ptr = std::shared_ptr<IDataProvider>;
+
+    virtual ~IDataProvider() {};
+    virtual size_t provide_data(size_t out_data_bytes_size, void* out_data) = 0;
+    virtual bool empty() const = 0;
```

Comment:
Does `empty` means data provider e.g. failed to open or does it signal EOF?

Reply:
`ctor` thows exception `DataProviderSystemErrorException` here. So after creation `DataProvider` must be valid and `empty()` means no more data

Reply:
will add Doxygen comment